google-oauth-client/src/main/java/com/google/api/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -447,23 +447,41 @@ public final Clock getClock() {
    */
   private static class PKCE {
     private final String verifier;
-    private final String challenge;
+    private String challenge;
     private String challengeMethod;
 
     public PKCE() {
       verifier = generateVerifier();
-      Challenge c = new Challenge(verifier);
-      challenge = c.challenge;
-      challengeMethod = c.method;
+      generateChallenge(verifier);
     }
 
-    private String generateVerifier() {
+    private static String generateVerifier() {
       SecureRandom sr = new SecureRandom();
       byte[] code = new byte[32];
       sr.nextBytes(code);
       return Base64.encodeBase64URLSafeString(code);
     }
 
+    /**
+     * Create the PKCE code verifier. It will use the S256 method but
```

Comment:
nit: present tense per Google style
i.e. will use the --> uses the
will fall back --> falls back